File: IQCare.Sync/IQCare.Sync.Core/Interfaces/Repository/IRepository.cs
using System;
using System.Collections.Generic;

namespace IQCare.Sync.Core.Interfaces.Repository
{
    public interface IRepository<T>
    {
        T FindById(int id);
        T FindBySyncId(Guid uuid);
        IEnumerable<T> GetAll();
        void Save(T entity);
        void Save(List<T> entities);
        void Update(T entity);
        void Update(List<T> entities);
        void Delete(Guid uuid);
    }
}

File: IQMobileApp/core/src/main/java/com/thepalladiumgroup/iqm/core/data/IRepository.java
package com.thepalladiumgroup.iqm.core.data;

/**
 * Created by Koske Kimutai [2016/04/19]
 */

import java.sql.SQLException;
import java.util.List;

/**
 * Created by Koske Kimutai [2016/04/14]
 */
public interface IRepository<T> {
    T refresh(T entity) throws SQLException;
    T find(int id) throws SQLException;

    T findbyfield(String field, Object value) throws SQLException;

    T findbySyncId(String uuid) throws SQLException;

    T save(T entity) throws SQLException;

    void saveOnly(T entity) throws SQLException;

    T update(T entity) throws SQLException;

    void updateOnly(T entity) throws SQLException;

    void delete(int id) throws SQLException;

    void delete(T entity) throws SQLException;

    void deleteAll() throws SQLException;

    void deleteAll(List<T> entities) throws SQLException;
    List<T> getAll() throws SQLException;

    List<T> getAllbyfield(String field, Object value) throws SQLException;

}